extensions/control-plane/api/management-api/contract-definition-api/src/main/java/org/eclipse/edc/connector/api/management/contractdefinition/transform/JsonObjectToContractDefinitionTransformer.java

```diff
@@ -36,19 +37,23 @@ public JsonObjectToContractDefinitionTransformer() {
     @Override
     public @Nullable ContractDefinition transform(@NotNull JsonObject object, @NotNull TransformerContext context) {
         var builder = ContractDefinition.Builder.newInstance();
-
         builder.id(nodeId(object));
-
-        visitProperties(object, (key, value) -> {
-            switch (key) {
-                case CONTRACT_DEFINITION_ACCESSPOLICY_ID -> builder.accessPolicyId(transformString(value, context));
-                case CONTRACT_DEFINITION_CONTRACTPOLICY_ID -> builder.contractPolicyId(transformString(value, context));
-                case CONTRACT_DEFINITION_ASSETS_SELECTOR -> builder.assetsSelector(transformArray(value, Criterion.class, context));
-                default -> { }
-            }
-        });
-
+        visitProperties(object, (s, jsonValue) -> transformProperties(s, jsonValue, builder, context));
         return builderResult(builder::build, context);
     }
 
+    private void transformProperties(String key, JsonValue jsonValue, ContractDefinition.Builder builder, TransformerContext context) {
+        if (CONTRACT_DEFINITION_ACCESSPOLICY_ID.equals(key) ) {
```

Comment:
I think it could be motivated by the additional condition `jsonValue instanceof JsonArray` but that's useless: in an expanded jsonld document every property is either an array or null.

Reply:
This is addresses